### PR TITLE
Removed component-specific stuff from Model.js

### PR DESCRIPTION
**Pull request by vlbe**
First step to remove model.js

### CODE REVIEW

**Comment by chrisknoll** [comment]
Great work. This is a very sensitive/fragile class so we'll have to make sure that we have very deep testing here because the methods used in Model.js are not obviously linked to any specific feature in Atlas...but once we get this squared away, it will be much cleaner.

File: js/pages/jobs/job-manager.js [hunk 1]
```diff
@@ -1,66 +1,72 @@
...
+      this.jobs = ko.observableArray([]);
+      this.columns = ko.observableArray([
+        { title: 'ExecutionId', data: 'executionId' },
+        { title: 'Job Name', data: 'jobParameters.jobName' },
+        { title: 'Status', data: 'status' },
+        { title: 'Start Date', data: 'startDate', type: 'datetime-formatted' },
+        { title: 'End Date', data: 'endDate', type: 'datetime-formatted' },
+      ]);
+      if (config.userAuthenticationEnabled) {
+        this.columns.splice(3, 0, {
```
Comment (pavgra) [inline js/pages/jobs/job-manager.js]: Let's remove magic values (either create constants or calculate the numbers)

File: js/pages/configuration/roles/role-details.js [hunk 2]
```diff
@@ -1,333 +1,329 @@
...
+          ? authApi.isPermittedCreateRole()
+          : authApi.isPermittedReadRole(this.roleId());
+      });
+      this.canEditRole = ko.pureComputed(() => {
+        return this.isAuthenticated() && this.isNewRole()
+          ? authApi.isPermittedCreateRole()
+          : authApi.isPermittedEditRole(this.roleId());
+      });
+      this.canEditRoleUsers = ko.pureComputed(() => {
+        return (
```
Comment (pavgra) [inline js/pages/configuration/roles/role-details.js]: Let's not mix refactoring and code formatting changes - hard to review

File: js/pages/configuration/routes.js [hunk 3]
```diff
@@ -27,7 +29,7 @@ define(
         '/role/:id': new AuthorizedRoute((id) => {
           appModel.activePage(this.title);
           require(['./roles/role-details'], function () {
-            appModel.currentRoleId(id);
+            atlasState.ConfigurationRole.selectedId(id !== 0 ? id : null)
```
Comment (pavgra) [inline js/pages/configuration/routes.js]: Are you sure that the zero is numeric, not string here? I would add `parseInt`

File: js/pages/configuration/sources/source-manager.js [hunk 4]
```diff
@@ -300,62 +274,78 @@ define([
...
-            return roleService.getList();
-        })
-        .then((roles) => {
-          this.model.roles(roles);
-          this.loading(false);
-          this.goToConfigure();
-        })
-        .catch(() => this.loading(false));
+
+      try {
```
Comment (pavgra) [inline js/pages/configuration/sources/source-manager.js]: Let's not mix refactoring and code syntax changes - hard to review

**Pull request closed without merging**